app/src/main/java/com/mapbox/vision/examples/ScoreProgressView.kt

```diff
@@ -0,0 +1,203 @@
+                100 -> gradientColors[0]
+                in SCORE_MIN..(SCORE_MAX / 2) -> {
+                    argbEvaluator.evaluate(
+                        (currentScore.toFloat() / (SCORE_MAX / 2)),
+                        gradientColors[2],
+                        gradientColors[1]
+                    ) as Int
+                }
+                in (SCORE_MAX / 2)..SCORE_MAX -> {
+                    argbEvaluator.evaluate(
+                        ((currentScore - 50).toFloat() / (SCORE_MAX / 2)),
+                        gradientColors[1],
+                        gradientColors[0]
+                    ) as Int
+
+                }
+                else -> gradientColors[1]
+            }
+
+
+    private val progressGradientWidth by lazy(LazyThreadSafetyMode.NONE) { progress_gradient.width }
+
+    companion object {
+        private const val SCORE_MIN = 0
+        private const val SCORE_MAX = 100
+    }
+
+    init {
+        View.inflate(context, R.layout.score_progress_view, this)
+        progressViewContainer.bringToFront()
+        progress_score_container.bringToFront()
+
+        val gradientDrawable = GradientDrawable(GradientDrawable.Orientation.LEFT_RIGHT, gradientColors)
+            .apply {
+                cornerRadius = context.dpToPx(3f)
+            }
+        progress_gradient.background = gradientDrawable
+
+        setupInitScoreMark()
+    }
+
+    private fun setupInitScoreMark() {
+        val markFactor = 1 - initScore.toFloat() / SCORE_MAX
+        init_score_mark.layoutParams =
+            (init_score_mark.layoutParams as ConstraintLayout.LayoutParams).apply { horizontalBias = markFactor }
+    }
+
+    fun plusScore(amount: Int) = onNewScore(DriverScore.Positive(amount))
+
+    fun minusScore(amount: Int, reason: String) = onNewScore(DriverScore.Negative(amount, reason))
+
+    private fun onNewScore(nextScore: DriverScore) {
+        val delta = (animationWillEndAt - System.currentTimeMillis()).let { if (it < 0) 0 else it }
+
+        val delayMills: Long = delta + betweenNewScoreDuration
+
+        animationWillEndAt = when (nextScore) {
+            is DriverScore.Positive -> 0
+            is DriverScore.Negative -> animationAppearDuration + animationDisappearDuration + warningAndScoreShowDuration
+        } + delta + System.currentTimeMillis()
+
+        handlerScore.postDelayed({
+
+            when (nextScore) {
+                is DriverScore.Positive -> {
+                    currentScore += nextScore.score
+                }
+                is DriverScore.Negative -> {
+                    currentScore -= nextScore.score
+                    showWarning(nextScore.reason)
+                    showWarningScore(nextScore.score)
+                }
+            }
+
+        }, delayMills)
+    }
+
+    private fun redrawProgress() {
+        progress_gray.layoutParams.width =
+            (progressGradientWidth * (1.01f - currentScore.toFloat() / SCORE_MAX)).toInt()
```

Comment:
1.01 ?

Reply:
yes, it's just to not check corner case when `progressGradientWidth*0`

Reply:
Do you think its worth to leave comment about it?

Reply:
refactored